ibridges/session.py

```diff
@@ -40,11 +41,14 @@ def __init__(self, irods_env: dict = None, irods_env_path: str = None,
             if os.path.isfile(irods_env_path):
                 with open(irods_env_path, "r", encoding="utf-8") as f:
                     irods_env = json.load(f)
+                if not isinstance(irods_env, dict):
+                    raise ValueError(f"Error reading environment file '{irods_env_path}': "
+                                      "expected dictionary, got {irods_env}.")
             else:
                 raise ValueError(f"CONNECTION ERROR: {irods_env_path} path does not exist.")
 
         self._password = password
-        self._irods_env = irods_env
+        self._irods_env: dict = irods_env  # type: ignore
```

Comment:
Just for my understanding: is this required because at this point `self._irods_env` is no longer Optional?

Reply:
Yes, this helps MyPy to understand that it is a dictionary. I wasn't able to make it understand that the logic is sound and that it will always become a dict, so that is why it is ignored.